python/whylogs/core/schema.py

```diff
@@ -107,6 +110,7 @@ def copy(self) -> "DatasetSchema":
         args = {k: deepcopy(self.__dict__[k]) for k in keys if k not in self.types}
         copy = self.__class__(**args)
         copy._columns = deepcopy(self._columns)
+        copy.segments = self.segments.copy()
```

Comment:
Does this need to be a deep copy like Dict and Columns?

Reply:
If we wanted to support special casing segmentation in subclasses of DatasetSchema, but I don't think we want to support that scenario.